libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/speedlimit/SpeedLimitWidget.kt

```diff
@@ -26,68 +28,87 @@ class SpeedLimitWidget(
     /**
      * The horizontal margin of the widget relative to the map.
      */
-    marginX: Float = 26f,
+    marginX: Float = 14f,
     /**
      * The vertical margin of the widget relative to the map.
      */
-    marginY: Float = 50f,
-) : BitmapWidget(drawSpeedLimitSign(speedLimit = null, speed = 0), position, marginX, marginY) {
+    marginY: Float = 30f,
+    /**
+     * The initial sign format to use before speed limit info is available.
+     */
+    initialSignFormat: SpeedLimitSign = SpeedLimitSign.MUTCD,
+) : BitmapWidget(
+    drawSpeedLimitSign(speedLimit = null, speed = 0, initialSignFormat, warn = false),
+    position, marginX, marginY,
+) {
 
     private var lastSpeedLimit: Int? = null
     private var lastSpeed = 0
+    private var lastSignFormat = initialSignFormat
+    private var lastWarn = false
 
-    fun update(speedLimit: Int?, speed: Int) {
-        if (lastSpeedLimit == speedLimit && lastSpeed == speed) return
+    fun update(speedLimit: Int?, speed: Int, signFormat: SpeedLimitSign?, threshold: Int) {
+        val newSignFormat = signFormat ?: lastSignFormat
+        val warn = speedLimit != null && speed - threshold >= speedLimit
+        if (lastSpeedLimit == speedLimit &&
+            lastSpeed == speed &&
+            lastSignFormat == newSignFormat &&
+            lastWarn == warn
+        ) return
         lastSpeedLimit = speedLimit
         lastSpeed = speed
+        lastSignFormat = newSignFormat
+        lastWarn = warn
+
+        updateBitmap(drawSpeedLimitSign(speedLimit, speed, newSignFormat, warn))
+    }
+
+    fun update(signFormat: SpeedLimitSign?, threshold: Int) {
+        val speedLimit = lastSpeedLimit
+        val speed = lastSpeed
+        val newSignFormat = signFormat ?: lastSignFormat
+        val warn = speedLimit != null && speed - threshold >= speedLimit
+        if (lastSignFormat == newSignFormat && lastWarn == warn) return
+        lastSignFormat = newSignFormat
+        lastWarn = warn
 
-        updateBitmap(drawSpeedLimitSign(speedLimit, speed))
+        updateBitmap(drawSpeedLimitSign(speedLimit, speed, newSignFormat, warn))
     }
 
     internal companion object {
         private const val TAG = "SpeedLimitWidget"
-        private const val WIDTH = 55
-        private const val HEIGHT = 98
-        private const val PADDING = 3f
-        private const val STROKE = 2f
-        private const val RADIUS = 10f
-        private const val SIGN_HEIGHT = 57
+        private const val WIDTH_VIENNA = 74
```

Comment:
Nitpicking some pixels, but the top of the sign looks crammed.

IMO we should add some space between the top of the sign, and the "speed" text

<img width="293" alt="Screen Shot 2022-09-06 at 2 01 27 PM" src="https://user-images.githubusercontent.com/3021882/188737333-f886ec23-30d8-4ea9-90e2-2a9c821540cc.png">

Reply:
There's a bit more space on the regular sign

![Screen Shot 2022-09-06 at 2 02 21 PM](https://user-images.githubusercontent.com/3021882/188737509-fa543089-9bb0-41b1-827e-a3416c9f7340.png)

Reply:
Thank you, there was an error in paddings, this should be fixed now.